Bonsai.Core/Expressions/ExpressionBuilderGraphExtensions.cs

```diff
@@ -1023,6 +1023,31 @@ public static IObservable<Unit> BuildObservable(this ExpressionBuilderGraph sour
             return observableFactory();
         }
 
+        /// <summary>
+        /// Builds and compiles an expression builder workflow into an observable sequence
+        /// with the specified element type.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the elements in the observable sequence.</typeparam>
+        /// <param name="source">The expression builder workflow to compile.</param>
+        /// <returns>
+        /// An observable sequence with the specified element type.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// The specified expression builder workflow does not compile into an observable sequence
+        /// with the expected element type.
+        /// </exception>
+        public static IObservable<TResult> BuildObservable<TResult>(this ExpressionBuilderGraph source)
+        {
+            var workflow = source.Build();
+            if (!typeof(IObservable<TResult>).IsAssignableFrom(workflow.Type))
+            {
+                workflow = ExpressionBuilder.CoerceMethodArgument(typeof(IObservable<TResult>), workflow);
```

Comment:
Since this method is used for more than just arguments/parameters, would it make sense to rename `CoerceMethodArgument` to something more general like `CoerceType` or `CoerceExpression`?

It doesn't seem to be specific to arguments or argument-like things. (It's also used for properties elsewhere.)

Reply:
Yup, it started like that and then it proved itself too useful not to use 😅 it's fully internal so makes complete sense to rename.